### PR TITLE
Use Id<TagId> for tags instead of Strings

### PR DESCRIPTION
**Pull request by fabricedesre**
Fixing https://github.com/fxbox/taxonomy/issues/20

I added macros to reduce the boiler plate of Id creation, only to fail to use it in the doc test (could not figure out how to get the macro in scope there).

r? @Yoric

### CODE REVIEW

File: src/selector.rs [hunk 1]
```diff
@@ -28,9 +28,10 @@ pub trait SelectedBy<T> {
 /// ```
 /// use foxbox_taxonomy::selector::*;
 /// use foxbox_taxonomy::services::*;
+/// use foxbox_taxonomy::util::Id;
 ///
 /// let selector = ServiceSelector::new()
-///   .with_tags(vec!["entrance".to_owned()])
+///   .with_tags(vec![Id::<TagId>::new("entrance".to_owned())])
```
Comment (Yoric) [inline src/selector.rs:34]: `Id::new` should be sufficient here.

File: src/util.rs [hunk 2]
```diff
@@ -74,7 +72,6 @@ impl<T> Default for Phantom<T> {
     }
 }
 
-#[allow(new_without_default)]
```
Comment (Yoric) [inline src/util.rs:77]: Why did you remove that? Clippy keeps bothering me about the absence of a `Default`, even though there is one.

**Comment by Yoric** [comment]
r+ with a minor docnit and a question

File: src/selector.rs (hunk 1 above)
Comment (fabricedesre) [inline src/selector.rs:34]: No, the compiler complains : "error: unable to infer enough type information about `_`; type annotations or generic parameter binding required [E0282]"

File: src/util.rs (hunk 2 above)
Comment (fabricedesre) [inline src/util.rs:77]: It's not needed anymore with the rustc I set in .travis.yml
